src/plugins/share/public/components/url_panel_content.tsx

```diff
@@ -91,6 +101,46 @@ export class UrlPanelContent extends Component<Props, State> {
     this.setUrl();
 
     window.addEventListener('hashchange', this.resetUrl, false);
+
+    if (this.props.anonymousAccess) {
+      (async () => {
+        const anonymousAccessParameters = await this.props.anonymousAccess!.getAccessURLParameters();
+
+        if (!this.mounted) {
+          return;
+        }
+
+        if (!anonymousAccessParameters) {
+          return;
+        }
+
+        this.setState({
+          anonymousAccessParameters,
+        });
+      })();
+
+      if (this.props.showPublicUrlSwitch) {
+        (async () => {
+          const anonymousUserCapabilities = await this.props.anonymousAccess!.getCapabilities();
```

Comment:
__issue:__ this is one makes an HTTP call under the hood that it's only needed if `anonymousAccessParameters` isn't null. And the majority of the setups won't have anonymous access, so we can save some bandwidth here. How about something like this (not sure about the role of `mounted`, so may not use it properly)?
```ts
const { anonymousAccess, showPublicUrlSwitch } = this.props;
if (anonymousAccess && showPublicUrlSwitch) {
  (async () => {
    // [Oleg]: this should resolve almost immediately
    const anonymousAccessParameters = await anonymousAccess.getAccessURLParameters();
    if (!this.mounted || !anonymousAccessParameters) {
      return;
    }

    this.setState({
      anonymousAccessParameters,
      showPublicUrlSwitch: await showPublicUrlSwitch(await anonymousAccess.getCapabilities()),
    });
  })();
}
```

Reply:
I've moved the `getCapabilities` under the `anonymousAccessParameters` check, so HTTP request will not be executed if `anonymousAccessParameters` is `null`.